File: typescript-examples/playwright-basics-ts/api/01-basic-navigation.ts
import { BrowserContext, Page } from "playwright";

/**
 * Basic Navigation
 *
 * Demonstrates the simplest Playwright operation: navigating to a URL
 * and returning the page title.
 */

interface Params {
  url: string;
}

export default async function handler(
  params: Params,
  page: Page,
  context: BrowserContext
) {
  // Navigate to the URL
  await page.goto(params.url);

  // Get basic page info
  const title = await page.title();
  const url = page.url();

  return {
    title,
    url,
  };
}

File: python-examples/playwright-basics/api/01-basic-navigation.py
"""
Basic Navigation

Demonstrates fundamental page navigation:
- Navigating to URLs with goto()
- Getting page information (title, URL)
"""

from typing import TypedDict

from playwright.async_api import Page


class Params(TypedDict):
    url: str


async def automation(page: Page, params: Params | None = None, **_kwargs):
    url = (
        params.get("url", "https://books.toscrape.com/")
        if params
        else "https://books.toscrape.com/"
    )

    await page.goto(url)

    title = await page.title()
    current_url = page.url

    return {
        "title": title,
        "url": current_url,
    }
